testdata/p4_16_samples/psa-portid-using-newtype2.p4

```diff
@@ -0,0 +1,293 @@
+#include "core.p4"
+
+// From psa.p4
+
+#define PORT_BITWIDTH 9  // target-specific
+
+// PortIdUInt_t is only intended for casting values from type PortId_t
+// to an unsigned integer with the same width.
+typedef bit<PORT_BITWIDTH> PortIdUInt_t;
+
+// request translation by P4Runtime agent of all occurrences of PortId_t exposed
+// to control-plane
+@p4runtime_translate("port")
+newtype bit<PORT_BITWIDTH> PortId_t;
+
+//const PortId_t PSA_CPU_PORT = (PortId_t) 9w192; // target-specific
```

Comment:
Note 1: Even if the control plane conversion procedure involves value translation, this constant must be the *translated* value. The compiler will have no idea of the control-plane translation rules anyway. 

Note 2: It is precisely of the Note 1 that we want to put a cast to PortId_t (or any newtype-derived-type) in front of constants: because we want to ensure that the programmer understands that he/she must translate the values themselves (or that they are appropriate to use for other operations). 

It might be easier to think of what is required if you think of a control plane as a CLI that uses strings to represent different values of PortId_t.

Reply:
The '// target-specific' comment on that line is intended to convey that every target can define its own custom data-plane-specific value for PSA_CPU_PORT.  For any one given target, this value is a constant.  Across different targets, different constants are expected to be used.  That line belongs in a target-specific psa.p4 include file.  Sorry if that wasn't clear in the example code.